external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -225,6 +237,23 @@ private long doSeek(TopicPartition tp, OffsetAndMetadata committedOffset) {
         }
     }
 
+    /**
+     * Checks If {@link OffsetAndMetadata} was committed by this topology, either by this or another spout instance.
```

Comment:
Nit: Something like "Checks if OffsetAndMetadata was committed by a KafkaSpout in this topology." might be clearer.

Reply:
Also we might want to add notes to EARLIEST and LATEST to clarify that they only restart when the topology is redeployed.

Reply:
You mean EARLIEST, LATEST only apply when a topology (with a given id) is started for the first time, and not when it is activated/deactivated?

Reply:
Sort of. I believe the storm id will be new every time you deploy a topology. According to the javadoc for getStormId it's the topology name + a nonce. So EARLIEST and LATEST will apply when you deploy the topology, but not when activating/deactivating the topology, and it won't apply if the spout crashes either.

Reply:
We said the same thing and that is the intended behavior

Reply:
Great. We might still want to expand the javadoc for the constants in KafkaSpoutConfig, because right now it doesn't say anything about when the setting has an effect, just that it applies to "the first poll".